torch/csrc/distributed/c10d/comm.h

```diff
@@ -35,45 +35,33 @@ class GradBucket {
   std::vector<at::Tensor> tensors_;
 };
 
-// DDP's c10d reducer allows communication hooks defined as a sub class
-// of CommHookInterface. CommHookInterface is an abstract class and can
-// be used to implement both Python and CPP hooks.
-struct TORCH_PYTHON_API CommHookInterface {
+// Base class of both `PythonCommHook` and `CppCommHook`.
+// Requires implementing `runHook` method thhat communicate gradients
+// asynchronously.
+class TORCH_API CommHookInterface {
  public:
   virtual ~CommHookInterface() {}
 
-  // runHook takes a GradBucket type bucket and passes the tensors of
-  // this grad bucket to hook's callback. This function is called once
-  // the bucket is ready. The hook can perform whatever processing is
-  // needed and return a Future that will hold the new value of the grad
-  // bucket's tensors once ready.
+  // Runs the registered communication hook to communicate gradients
+  // asynchronously, Returns a future that holds the communication results.
   virtual c10::intrusive_ptr<torch::jit::Future> runHook(
```

Comment:
Should we preserve the old comments as well? They provide useful information about the hook's contract?

Reply:
Feel the old comments are a bit wordy, so rephrased it from 5 lines to 3 lines, which should not really lose any information.